pages/movies/[id].js

```diff
@@ -61,6 +60,14 @@ export default function MovieDetailsPage({ onSubmit, movieInfo }) {
     (item) => item.id === movie.id
   )?.reviews;
 
+  const averageRating = selectedReview
+    ?.map((review) => review.rating)
+    .reduce(
+      (accumulator, currentValue, index, array) =>
+        accumulator + currentValue / array.length,
+      0
```

Comment:
this code works so you can keep it, the "standard" way to do this would be to calculate the sum of the reviews first and then divide by the length of the array at the end: 
```js
const ratingSum = ratings.reduce((sum,rating) => sum + rating, 0)
const average = ratingSum / ratings.length
```